### DISCUSSION TITLE
how to use NAFF in matlab AT

### DISCUSSION THREAD

**Discussion by simoneliuzzo** [atcollab/at#544] [Matlab AT]
Which commands should be used in matlab AT to use NAFF algorithm to assess tunes from turn-by-turn data?

**Comment by swhite2401** [comment]
Hello @simoneliuzzo , for matlab I cannot comment but I remember there were issues with the license.
Alternatively there is the module `harmonic_analysis.py` of pyAT that implements a same algorithm as NAFF (`method=laskar`) to determine the tune

**Reply by simoneliuzzo** [reply]
Dear @swhite2401 ,

thank you for your reply. I would need the matlab version as the code I am running is written in matlab.

**Comment by carmignani** [comment]
Hello Simone,
there is the function calcnaff, written by @lnadolski.

**Reply by simoneliuzzo** [reply]
thank you @carmignani !

**Reply by lfarv** [reply]
Hi all,

Historically, NAFF was method 4 in `findtunes.m`. At some point, it was removed, I don't know why. The git history is not deep enough to find the original version, but I think it should be restored.

A search for `nafflib` (the mex function) in the entire project shows only `calcnaff`, so for the time being Nicola is right, it's the only access.
